includes/utilities.inc

```diff
@@ -438,15 +454,33 @@ function islandora_solr_truncate_field_display($display_values, $max_length, $ad
         break;
       }
     }
+
+    $original_value = implode($separator, $updated_display_values);
+    if (isset($link_options['link_to_object']) && $link_options['link_to_object'] == TRUE) {
+      $original_value = l($original_value, $link_options['url'], $link_options['options']);
+    }
+    elseif (isset($link_options['link_to_search']) && $link_options['link_to_search'] == TRUE) {
+      $solr_query = format_string('!field:"!value"', array(
+        '!field' => $link_options['field'],
+        '!value' => islandora_solr_lesser_escape($original_value),
+        ));
+      $original_value = l(
+        $original_value,
+        "islandora/search/$solr_query",
+        array(
+          'html' => TRUE,
+        )
+      );
+    }
     $updated_display_values
-      = '<span class="toggle-wrapper">' .
+      = "<span class='toggle-wrapper'>" .
       t("<span>!value !separator<a href='#' class='toggler'>Show more</a></span>", array(
         '!separator' => $separator,
         '!value' => implode($separator, $truncated_list),
```

Comment:
In this case the original truncated portion isn't linking out to the object where the "show more" version is.